File: wiki_java/src/main/java/com/github/klee0kai/wiki/cachecontrol/PlanetsModule.java
package com.github.klee0kai.wiki.cachecontrol;

import com.github.klee0kai.stone.annotations.module.BindInstance;
import com.github.klee0kai.stone.annotations.module.Module;
import com.github.klee0kai.test.mowgli.galaxy.Mercury;

@Module
public interface PlanetsModule {

    @GcMercuryScope
    @BindInstance
    Mercury mercury(Mercury sirius);

}

File: wiki_kotlin/src/main/kotlin/com/github/klee0kai/wiki/cachecontrol/PlanetsModule.kt
package com.github.klee0kai.wiki.cachecontrol

import com.github.klee0kai.stone.annotations.module.BindInstance
import com.github.klee0kai.stone.annotations.module.Module
import com.github.klee0kai.test.mowgli.galaxy.Mercury

@Module
interface PlanetsModule {

    @GcMercuryScope
    @BindInstance
    fun mercury(sirius: Mercury? = null): Mercury?

}
